### PR TITLE
Add JSON log iterator with functionality matching Framer class

### PR DESCRIPTION
**Pull request by pmiettinen**
Enabler for https://github.com/swift-nav/piksi_tools/pull/1137

### CODE REVIEW

File: python/sbp/client/handler.py [hunk 1]
```diff
@@ -62,15 +62,15 @@ def gen_messages():
 
         messages = gen_messages()
         while True:
```
Comment (denniszollo) [inline python/sbp/client/handler.py]: there are no changes to this file, so can we try and avoid changing whitespace?

**Comment by denniszollo** [comment]
Can you comment as to how this is different than the existing JsonLogIterator and the reasoning behind needing a new iterator rather than updating the existing JsonLogIterator?  I couldn't tell by inspecting the code.

I think we tried to make the existing iterator capable of handling all three types of json serialized SBP; but perhaps that was a bad idea.

File: python/sbp/client/handler.py (hunk 1 above)
Comment (denniszollo) [reply python/sbp/client/handler.py]: oh I see now that you renamed the variable.

**Comment by pmiettinen** [comment]
@denniszollo the main problem is the `yield` inside of `__next__`. It leads to a "we put an iterator into your iterator" kind of deal. I think it's somewhat mentioned in the class documentation also, see "Notes" here https://github.com/swift-nav/libsbp/blob/master/python/sbp/client/loggers/json_logger.py#L128

**Comment by denniszollo** [comment]
replace with #779 due to merge conflict

**Comment by silverjam** [comment]
closing since #779 was merged

**Pull request closed without merging**